react/daily-art-prompt/src/store/rootReducer.test.js

```diff
@@ -1,573 +1,506 @@
-            })
-            describe('when there are previous comments', () => {
-                it('should return image with new comment added', () => {
-                    const state = [
-                        {
-                            id: 1,
-                            comments: [{ id: 1 }]
-                        }
-                    ]
-                    const action = {
-                        type: TYPES.ADD_COMMENT,
-                        payload: {
-                            imageId: 1,
-                            value: 'new comment'
-                        }
-                    }
-                    expect(promptImagesReducer(state, action)).toEqual(
-                        [
-                            {
-                                id: 1,
-                                comments: [
-                                    {
-                                        id: 1
-                                    },
-                                    {
-                                        id: 2,
-                                        text: 'new comment'
-                                    }]
-                            }
-                        ]
-                    )
-                })
-            })
-        })
-        describe('when image.id does not equal payload.imageId', () => {
-            it('should return image unchanged', () => {
-                const state = [{
-                    id: 1,
-                    comments: []
-                }]
-                const action = {
-                    type: TYPES.ADD_COMMENT,
-                    payload: {
-                        imageId: 2,
-                        value: 'new comment'
-                    }
-                }
-                expect(promptImagesReducer(state, action)).toEqual(state)
-            })
-        })
-    })
+describe("promptImagesReducer", () => {
+  describe("when action.type equals SET_INITIAL_IMAGES", () => {
+    describe("when there is no images in state", () => {
+      it("should return updatedImages with correct attributes", () => {
+        const state = [];
+        const action = {
+          type: TYPES.SET_INITIAL_IMAGES,
+          payload: {
+            image: "some image"
+          }
+        };
+        expect(promptImagesReducer(state, action)).toEqual(["some image"]);
+      });
+    });
+    describe("when there are images in state", () => {
+      it("should return updatedImages with previous state and new Image", () => {
+        const state = [
+          {
+            id: 2,
+            src: "firstImage",
+            liked: false,
+            comments: []
+          }
+        ];
+        const action = {
+          type: TYPES.SET_INITIAL_IMAGES,
+          payload: {
+            image: "apiCall"
+          }
+        };
+        expect(promptImagesReducer(state, action)).toEqual([
+          ...state,
+          "apiCall"
+        ]);
+      });
+    });
+  });
 
```

Comment:
what happens if the image provided is null or undefined?

Reply:
the ground splits open and Armageddon begins!

but I added the test and then added an if statement to catch the condition if it is null/undefined